packages/legend-application-studio/src/stores/EditorTabManagerState.ts

```diff
@@ -93,44 +91,64 @@ export class EditorTabManagerState extends TabManagerState {
    * cause memory issues
    *
    * See https://github.com/finos/legend-studio/issues/1713
-   *
-   * FIXME: we allow this so the UX stays the same but this can cause memory leak
-   * we should change `reprocess` model to do something like having source information
-   * on the form to navigate to it properly so that information is not dependent on the
-   * graph, but on the component itself, with IDs and such.
-   *
-   * TODO: to be removed when we process editor states properly
-   *
-   * @risk memory-leak
    */
-  recoverTabs(tabs: EditorState[], currentTab: EditorState | undefined): void {
-    let newCurrentTab: EditorState | undefined;
-    this.tabs = tabs
-      .map((tab) => {
-        let newTab: EditorState | undefined = undefined;
-        if (tab instanceof ElementEditorState) {
+  recoverTabs = (
+    openedTabEditorPaths: string[],
+    openedGeneratedFileTabStates: FileGenerationViewerState[],
+    currentTabState: EditorState | undefined,
+    currentTabElementPath: string | undefined,
+    shouldRecoverTabs: boolean,
+  ): void => {
+    const getElemenetPathFromFilePath = (filePath: string): string => {
```

Comment:
I don't think `getElemenetPathFromFilePath` captures everything we need to confirm if that file is still being used. 
What is the risk of just not reprocessing those files ? have the user re generate if they want those files to re appear ?

Reply:
there is no risk in reprocessing them according to me. They might just have to open them when they go in form mode.

Reply:
yeah i'd rather not than. i know we plan to do more in artifact generation extension and `getElemenetPathFromFilePath` dosesn't caputre everything